Titulo: Nueva Rama

demepty opened the pull request:
Creo nueva rama y agrego los cambioss hechos en los 2 ultimos sabados,
pruebas en desarrollo y conexion a mongo db

Los archivos modificados son:

 Changes to be committed:
	renamed:    server/bin/www -> bin/www
	deleted:    server/app.js
	modified:   server/package-lock.json
	modified:   server/package.json
	deleted:    server/routes/index.js
	deleted:    server/routes/users.js
	new file:   server/src/app.js
	new file:   server/src/models/Thread.js
	new file:   server/src/mongoose.js
	new file:   server/src/routes/threads.js
	new file:   server/test/integration/fixtures.js
        new file:   server/test/integration/mongooseHelper.js
	new file:   server/test/integration/threads.test.js

demepty committed:
Titulo: Nueva Rama

Creo nueva rama y agrego los cambioss hechos en los 2 ultimos sabados,
pruebas en desarrollo y conexion a mongo db

Los archivos modificados son:

 Changes to be committed:
	renamed:    server/bin/www -> bin/www
	deleted:    server/app.js
	modified:   server/package-lock.json
	modified:   server/package.json
	deleted:    server/routes/index.js
	deleted:    server/routes/users.js
	new file:   server/src/app.js
	new file:   server/src/models/Thread.js
	new file:   server/src/mongoose.js
	new file:   server/src/routes/threads.js
	new file:   server/test/integration/fixtures.js
        new file:   server/test/integration/mongooseHelper.js
	new file:   server/test/integration/threads.test.js

demepty committed:
Titulo: Realizo los cambios indicados

Realice los cambios solicitados y desarrrollados en la clase del 7 de
abril sobre los test y los del dia de hoy sobre redux.

No se pudo instalar redux por problemas con la pagina.
Los archivos modificados son:

 Changes to be committed:
	new file:   client/src/pages/ThreadsPage/reducer.js
	modified:   server/package.json
	modified:   server/src/app.js
	modified:   server/src/mongoose.js
	modified:   server/test/integration/fixtures.js
	modified:   server/test/integration/threads.test.js

christopher-francisco commented on client/src/pages/ThreadsPage/reducer.js:
```diff
@@ -0,0 +1,27 @@
+const initialState = {
+threads:[],
```
La indentación debe ser de 2 espacios.

christopher-francisco commented on client/src/pages/ThreadsPage/reducer.js:
```diff
@@ -0,0 +1,27 @@
+};
+
+export default (state = initialState, action) => {
+
```
Borra esta linea en blanco

christopher-francisco commented on client/src/pages/ThreadsPage/reducer.js:
```diff
@@ -0,0 +1,27 @@
+      break;
+  }
+
+
```
Borra estas lineas en blanco

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+var logger = require('morgan');
+var bodyParser = require('body-parser');
+
+
```
Borra esta línea en blanco

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+}
+app.use(logger('dev'));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
```
Borra esta linea, no estamos usando `urlencoded`

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+//app.use(cookieParser());
+//app.use(express.static(path.join(__dirname, 'public')));
+
+//app.use('/', index);
```
Borra todos estas lineas comentadas, y deja un espacio de segmentación lógica:

```
...
app.use(bodyParser.json());

app.use('/threads', threadsRouter);
...
```

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+
+// error handler
+app.use((err,req, res, next)=> {
+  // set locals, only providing error in development
```
borra este comment

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+  const message = err.message;
+  const stack = req.app.get('env') === 'development' ? err.stack : {};
+
+  // render the error page
```
borra este comment

christopher-francisco commented on server/src/app.js:
```diff
@@ -0,0 +1,45 @@
+    stack,
+    message
+  });
+  //res.render('error');
```
borra esta linea comentada

christopher-francisco commented on server/src/models/Thread.js:
```diff
@@ -0,0 +1,30 @@
+const Schema =mongoose.Schema;
+const ObjectId = Schema.Types.ObjectId;
+
+const ThreadSchema = new Schema ({
```
Borra el espacio entre `Schema` y `({`

christopher-francisco commented on server/src/models/Thread.js:
```diff
@@ -0,0 +1,30 @@
+    type:String,
+    lowercase:true,
+  },
+
```
borra esta linea en blanco

christopher-francisco commented on server/src/models/Thread.js:
```diff
@@ -0,0 +1,30 @@
+});
+
+ThreadSchema.pre('save',function(next){
+ this.slug = slug(this.title);
```
Indentación a 2 espacios en esta función

christopher-francisco commented on server/src/models/Thread.js:
```diff
@@ -0,0 +1,30 @@
+ThreadSchema.pre('save',function(next){
+ this.slug = slug(this.title);
+ return next();
+
```
borra esta linea en blanco

christopher-francisco commented on server/src/models/Thread.js:
```diff
@@ -0,0 +1,30 @@
+});
+
+const Thread = mongoose.model('Thread',ThreadSchema);
+module.exports = Thread;
```
Agrega una linea en blanco (para segmentar espacios logicos) entre `const ...` y `module.exports`

christopher-francisco commented on server/src/mongoose.js:
```diff
@@ -0,0 +1,9 @@
+const mongoose = require('mongoose');
+//TO_DO: Refactor
```
Agrega un espacio en blanco arriba para segmentar los modulos importados del resto del codigo

christopher-francisco commented on server/src/routes/threads.js:
```diff
@@ -0,0 +1,16 @@
+
+router.get('/', async (req,res)=>{
+  //res.send('hellow');
+
```
Borra la línea comentada, y la línea en blanco

christopher-francisco commented on server/src/routes/threads.js:
```diff
@@ -0,0 +1,16 @@
+  const threads = await Thread.find();
+
+  res.json({
+    data: threads ,
```
Borra el espacio entre `threads` y `,`

christopher-francisco commented on server/src/routes/threads.js:
```diff
@@ -0,0 +1,16 @@
+  });
+});
+
+
```
borra esta linea en blanco

christopher-francisco commented on server/test/integration/fixtures.js:
```diff
@@ -0,0 +1,22 @@
+const Thread = require ('../../src/models/Thread');
```
Arregla la indentación de todo este archivo, a 2 espacios. Cada subnivel debe tener 2 espacios adicionales al anterior:

```
primer nivel {
  segundo nivel {
    tercer nivel {
      ...
    }
  }
}
```

christopher-francisco commented on server/test/integration/fixtures.js:
```diff
@@ -0,0 +1,22 @@
+
+module.exports={
+seedThreads,
+
```
borra esta linea en blanco

christopher-francisco commented on server/test/integration/mongooseHelper.js:
```diff
@@ -0,0 +1,58 @@
+const mongoose = require('mongoose');
```
Corrige la indentación a 2 espacios, y las líneas en blanco, de todo este archivo

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+//Modulos para testing
+
```
borra estas 2 líneas

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+const request= require('supertest');
+const chai = require('chai');
+
+//Rutas de archivos de codigo fuente
```
borra este comentario

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+
+//Rutas de archivos de codigo fuente
+const mongooseHelper=require('./mongooseHelper');
+//como iniccializar la db para probar data de prueba
```
Este comentario puedes borrarlo. El concepto de "Fixtures" es conocido y siempre indica llenar datos en la base de datos para hacer pruebas.  Sin embargo, si lo quieres mantener, ponlo dentro de fixtures, en vez de aquí en el test.

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+const mongooseHelper=require('./mongooseHelper');
+//como iniccializar la db para probar data de prueba
+const fixtures = require('./fixtures');
+const app = require('../../src/app'); //2 pts representan una carpeta hacia arriba
```
Borra el comentario.

Lo de los 2 puntos `..` para subir una carpeta arriba, es conocimiento general de Linux.  Te recomiendo mantener este tipo de información en tus notas aparte, fuera del código fuente.

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+
+const expect = chai.expect;
+
+//funciona para realizar los test de manera legible
```
Borra este comentario.

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+describe('Threads end point ', () => {
+
+//before ( async () => await mongooseHelper.connectToDatabase());
+//after ( async () => await mongooseHelper.closeConnection());
```
Borra estas líneas comentadas

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+    describe('when data base has threads',()=>{
+      before ( async () => await fixtures.seedThreads());
+      after ( async () => await mongooseHelper.dropDatabase()); //limpiamos la bd
+      it('returns a list of threads', () => {  //esperamos que retorne este tipo de componente
```
Agrega una línea en blanco entre `after` y `it` para segmentación lógica; y borra los comentarios

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+
+    describe('when data base has threads',()=>{
+      before ( async () => await fixtures.seedThreads());
+      after ( async () => await mongooseHelper.dropDatabase()); //limpiamos la bd
```
Borra los comentarios. El nombre de la función "dropDatatabase" significa "borrar la base de datos", por lo que el comentario es redundante.

christopher-francisco commented on server/test/integration/threads.test.js:
```diff
@@ -0,0 +1,85 @@
+
+});
+
+
```
Arregla la indentación a 2 espacios por nivel, y las líneas en blanco de este archivo.

demepty commented:
Gracias Chris , hare los cambios . Saludos

demepty committed:
Titulo: Cree los actions y arregle indentacion

Cree las acciones y el test para las acciones , tambien mejore la
indentacion de los archivos segun los comentarios de Christopher, queda
pendiente realizar el archivo para test del reducer.

 Changes to be committed:
	new file:   client/src/pages/ThreadsPage/actions.js
	new file:   client/src/pages/ThreadsPage/actions.spec.js
	modified:   client/src/pages/ThreadsPage/reducer.js
	new file:   client/src/pages/ThreadsPage/reducer.spec.js
	modified:   server/src/app.js
	modified:   server/src/models/Thread.js
	modified:   server/src/mongoose.js
	modified:   server/src/routes/threads.js
	modified:   server/test/integration/fixtures.js
	modified:   server/test/integration/mongooseHelper.js
	modified:   server/test/integration/threads.test.js

demepty committed:
Cambios hasta el momento

Estos son los cambios que llevo hasta el momento.

Los ultimo archivos modificados son:
	modified:   client/src/pages/ThreadsPage/actions.js
	modified:   client/src/pages/ThreadsPage/reducer.js
	modified:   client/src/pages/ThreadsPage/reducer.spec.js
